clients/js/src/compat.ts

```diff
@@ -322,14 +236,6 @@ function wrapEthers6Provider<P extends ethers6.Provider>(
   return makeProxy(provider, cipher, hooks);
 }
 
-function isEthers5Provider(upstream: unknown): upstream is Ethers5Provider {
-  return AbstractProvider.isProvider(upstream);
-}
-
-function isEthers5Signer(upstream: unknown): upstream is Ethers5Signer {
-  return AbstractSigner.isSigner(upstream) && '_signTypedData' in upstream;
-}
-
 function isEthers6Signer(upstream: unknown): upstream is ethers6.Signer {
   return upstream instanceof ethers6.AbstractSigner;
```

Comment:
We ran into a bug where `instanceof` doesn't work well with NPM and potentially slightly different versions of ethers v6 being used by the package and being used by other modules in the same bundled project.

This is why with Ethers v5 we're using `AbstractSigner.isSigner` & `AbstractProvider.isProvider` which doesn't do direct `instanceof` comparison.

e.g.
```typescript
    static isProvider(value: any): value is Provider {
        return !!(value && value._isProvider);
    }
```

Reply:
Unfortunately these utility functions were removed in Ethers v6. https://docs.ethers.org/v6/single-page/#api_providers__Provider